src/solver/conv_hip_implicit_gemm_wrw_v4r4_xdlops.cpp

```diff
@@ -0,0 +1,1074 @@
+    result.invoker_factory = [=](const std::vector<Kernel>& kernels) {
+        return [=](const Handle& handle, const boost::any& primitive_params) {
+            const auto invoke_params = boost::any_cast<conv::WrWInvokeParams>(primitive_params);
+            const auto& tensors      = invoke_params.tensors;
+            auto kernel              = handle.Run(kernels[0]);
+            float elapsed            = 0;
+            float zero               = 0.f;
+            if(ctx.IsFp16() || ctx.IsBfp16())
+            {
+                const auto& workSpace = invoke_params.workSpace;
+                TensorDescriptor workspaceDesc(
+                    miopenFloat, tensors.dwDesc.GetLengths(), tensors.dwDesc.GetStrides());
+                SetTensor(handle, workspaceDesc, workSpace, &zero);
+                if(handle.IsProfilingEnabled())
+                {
+                    elapsed += handle.GetKernelTime();
+                }
+                kernel(tensors.x, tensors.dy, workSpace);
+                if(handle.IsProfilingEnabled())
+                {
+                    elapsed += handle.GetKernelTime();
+                }
+                CastTensor(handle,
+                           &lowp_quant,
+                           workspaceDesc,
+                           workSpace,
+                           tensors.dwDesc,
+                           tensors.dw,
+                           0,
+                           0);
+            }
+            else
+            {
+                SetTensor(handle, tensors.dwDesc, tensors.dw, &zero);
+                handle.Run(kernels[0])(tensors.x, tensors.dy, tensors.dw);
+            }
+
+            if(handle.IsProfilingEnabled())
+            {
+                elapsed += handle.GetKernelTime();
+                handle.ResetKernelTime();
+                handle.AccumKernelTime(elapsed);
+            }
+        };
+    };
+    result.workspce_sz = GetWorkspaceSize(ctx);
+    return result;
+}
+
+int ConvHipImplicitGemmWrwV4R4Xdlops::RunAndMeasureSolution(const miopen::Handle& profile_h,
+                                                            ConstData_t bot_buf,
+                                                            ConstData_t top_buf,
+                                                            Data_t wei_buf,
+                                                            ConstData_t bias_buf,
+                                                            const ConvolutionContext& ctx,
+                                                            const ConvSolution& solution,
+                                                            float& elapsed_time) const
+{
+    assert(bias_buf == nullptr);
+    (void)bias_buf;
+
+    return RunAndMeasureSolutionBase(
+        profile_h, bot_buf, top_buf, wei_buf, ctx, solution, elapsed_time);
+}
+
+bool ConvHipImplicitGemmWrwV4R4Xdlops::IsApplicable(const ConvolutionContext& ctx) const
+{
+    if(!IsXdlopsSupport(ctx))
```

Comment:
```C++
    if(ctx.skip_solutions_that_take_long_time_to_build_and_have_narrow_coverage)
        return false;
```

Reply:
This estimation would take much work to implement, since no one has done it before for HIP kernel, it's an unknown territory. These HIP static kernels are to increase roofline performance of xdlops fp16 performance, that will require tuning the whole network anyway.

For the network that need low compilation time, I imagine these kernels will be disabled any way.

So I suggest we don't implement this estimation in this PR.

Reply:
This estimation would take much work to implement, since no one has done it before for HIP kernel, it's an unknown territory. These HIP static kernels are to increase roofline performance of xdlops fp16 performance, that will require tuning the whole network anyway.

For the network that need low compilation time, I imagine these kernels will be disabled any way.

So I suggest we don't implement this estimation in this PR.

Reply:
@asroy Do you mean GetWti()? That is not what I meant in the review comment. I simply would like to insert two lines of code. Otherwise the FAST_HYBRID Find mode would be broken, see #326.

Reply:
Woops, I mis-understood it.

@ltqin Please add the 2 lines @atamazov suggested. It will skip HIP kernel during certain find mode.

Reply:
done